### DISCUSSION TITLE
Quais as melhores questões para abordar em um processo seletivo de frontend react Pleno e Senior?

### DISCUSSION THREAD

**Discussion by wilsonneto-swe** [frontendbr/forum#1762] [Discussão]
Fala galera, 
blz?

Quais julgam ser questões legais para abordar em entrevistas ou testes de recrutamento para frontend para trabalharem com react? Em uma aplicação de grande porte sendo refeita em SPA.

- ES 6+? Arrows functions, destructuring?
- Contextos do this em vários cenários?
- Patterns de projetos?
- Testar conhecimentos com Redux, Sagas, e Context?
- Testar conhecimentos específicos de libs como react-router-dom, axios ou outra para consumo de api?
- Na app atual não usamos TS, mas acham que seria legal abordar, já que no futuro vamos utilizar?

Mas a dúvida é, não sei bem o que aqui seria focado em um senior e o que seria focado em um pleno? Na vdd acho que dependo até um junior responderia a maioria das questões do framwork em si...

**Comment by deppbrazil** [comment]
@wilsonneto-dev cara vamos lá minha opinião sobre o que diferencia Junior, Pleno, Senior e Especialista.

* **Primeiro:** isso varia de empresas para empresas, projetos para projetos e times para times, mas vou tentar explicitar de modo geral, 

* **Segundo:** abordar esses ponto aqui que vc colocou em entrevistas talvez vc não consiga saber o melhor do candidato para níveis Pleno/Senior baseado na sua necessidade, pois deve ser levado em consideração a parte comportamental e momento de carreira também mesmo se tratando do bate-papo técnico na minha opinião ⏬

> * ES 6+? Arrows functions, destructuring?
> * Contextos do this em vários cenários?
> * Patterns de projetos?
> * Testar conhecimentos com Redux, Sagas, e Context?
> * Testar conhecimentos específicos de libs como react-router-dom, axios ou outra para consumo de api?
> * Na app atual não usamos TS, mas acham que seria legal abordar, já que no futuro vamos utilizar?

Pois mesmo que ele saiba de cabo a rabo esses pontos não garante que ele esteja em nível Pleno ou Senior para o mercado, gosto de dizer que senioridade vai além do código.

### O que eu acho sobre as posições:
* **Junior:** na minha opinião é o perfil que esta em fase de exploração, aqui pode se errar bastante (num bom sentido, tá?...rsrs) aqui o profissional por mais que entenda a fundo lógica de programação, já tenha trabalhado com outros frameworks ele vai ainda sentir dificuldade em tomar decisões para projetos grandes, ou talvez queira usar coisas hypes sempre e a realidade do projeto exige usar algo mais a moda antiga por questões estratégicas da companhia,
- > baseado na sua necessidade acima esse perfil seria legar ter: 
   - Bons conhecimento em Html, CSS e Javascript;
   - APIs;
   - Noções de Design;
   - Os fundamentos do ES6;
   - Já ter trabalhando com React e Redux;
   - Saber a importância dos testes;
   - ter trabalhado com CI/CD é um diferencial legal aqui;

* **Pleno:** na minha opinião, é o profissional com uma certa experiencia em Front-end no sentido de ter desenvolvido interfaces não apenas no cenário do React, e também que sim conheça padrões de Design, o básico de UX/UI, entenda também o lado do Back-end e Infra (entender não é ser FullStack, blz?), vai se preocupar muito com a performance do código (não que o jr não se preocupe, ok?), vai entender a necessidade dos testes unitários/integração/automatizado e saber a melhor hora para implementar caso o projeto não tenha, resumindo nos desafios do dia-a-dia por mais que tenha dificuldades em algo novo ele consegue sair do outro lado,
- > baseado na sua necessidade acima esse perfil seria legar ter: 
   - Sólidos conhecimento em Html, CSS e Javascript;
   - Conhecimento do lado do Back-end e Infra;
   - Acessibilidade na web;
   - API REST e outras arquiteturas como SOAP
   - Padrões de Design;
   - Noções de UX/UI;
   - Os fundamentos do ES6;
   - Já ter trabalhando com React e Redux;
   - Saber o que faz testes de integração, unidade e automatizado, se tiver trabalhado ou conhecer sobre: Jest, Enzyme, Cypress, React-testing-library, show de bola;
   - CI/CD;
   - Pipelines de integração, Infra e conhecimentos em DevOps é um diferencial

* **Senior:** aqui o profissional digo que é mais estratégico, ele entende a fundo a necessidade de negocio a ser aplicado no projeto, é um mentor para os demais devs, vai auxiliar no crescimento do time e na evolução do produto, consegue tomar e avaliar grandes decisões,
- > baseado na sua necessidade acima esse perfil seria legar ter: 
   - Conhecimento avançado em Html, CSS e Javascript;
   - Sólidos conhecimento do lado do Back-end e Infra;
   - Acessibilidade na web e seus padrões;
   - API REST e outras arquiteturas como SOAP
   - Padrões de Design;
   - Noções de UX/UI;
   - ES6
   - Já ter trabalhando com React e Redux;
   - Ter atuado com testes usando: Jest, Enzyme, Cypress, React-testing-library;
   - CI/CD;
   - Pipelines de integração, Infra, conhecimentos em DevOps;
   - Perfil comportamental no qual impulsiona o time e o produto
   - Ser um mentor para os demais devs;

* **Especialista:** aqui tem que conhecer de ponta-a-ponta o produto que está trabalhando, e principalmente todos os principais pontos no framework usado, entende muito a regra de negocio do produto, e como o Senior ele também é o mentor e busca sempre a evolução do time como um todo e toma grandes decisões para o produto/empresa,
- > baseado na sua necessidade acima esse perfil seria legar ter: 
   - Praticamente as mesmas características do Senior porém aqui é legal o profissional saber a fundo o React e todas as particularidade da lib, versões, etc e é bem assertivo nas tomadas de decisões de grandes impactos.
   - Aqui a experiência até em outras linguagens conta muito.

### Considerações finais:
* Esse é meu ponto de vista em processos seletivos de modo geral no mundo Front-end, visto que isso não esta escrito na pedra, é apenas minha opinião com base no que já vivi... 
* Um conselho que deixo é por mais que o produto ou time use React tente conhecer outras partes técnicas do candidato, visto que React é apenas +1 ferramenta, já vi muitos profissionais fazerem um curso completo de React e já se acharem Pleno para o mercado, em uma situação que ele precise contribuir em outro projeto que não usa React ai lascou...rsrs  😄
